### PR TITLE
feat: save mnemonic as keystore for signing transactions

### PR DESCRIPTION
**Pull request by kai0x**
What changed?
- allow user to import mnemonic for signing transactions
- saved as keystore and encrypted with password

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,6 +10,14 @@ npm install dispersed-cli
 
 ## Usage
 
+### Common commands
+
+```bash
+# saves wallet for signing transactions
+$ npx dispersed key import <mnemomic> <password>
```
Comment (kai0x) [inline README.md:17]: i'm going to do a separate pr later to change all these to armada. or when I set up the new repo for this.

note to self: also swap all the js to use typescript instead. really miss ts 😅

File: cli.js [hunk 2]
```diff
@@ -58,6 +60,28 @@ yargs(hideBin(process.argv))
             console.log(`Successfully created manifest: ${manifestPath}`);
         }
     )
+    .command(
+        'key import <mnemonic> <password>',
```
Comment (sbaranov) [inline cli.js:64]: FYI a good npm package for prompting for these would be "readline-sync"

File: cli.js [hunk 3]
```diff
@@ -58,6 +60,28 @@ yargs(hideBin(process.argv))
...
+        'Import a mnemonic into armada-cli. Mnemomic is not stored. A keystore is generated from mnemomic and used instead for signing armada transactions',
+        (yargs) => {
+            return yargs
+                .positional('mnemonic', {
```
Comment (sbaranov) [inline cli.js:68]: It should take PK instead of mnemonic since we don't need hierarchical accounts.

File: keystore.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+ * @param passw password to encrypt the keystore
+ */
+import { encryptKeystore } from "@ethersproject/json-wallets";
+import fs from "fs";
+import {ethers} from "ethers";
+
+export async function generateKeyStore(hdWallet, ith, passw) {
+  const wallet = hdWallet.derivePath(`m/44'/60'/0'/0/${ith}`);
+  const keyStore = await encryptKeystore(wallet, passw);
+  const filename = `keystore_${wallet.address}.json`;
```
Comment (sbaranov) [inline keystore.js:15]: We should put in in ~/.armada/keystore.json

Comment (sbaranov) [inline keystore.js:15]: I agree it's a good idea to support multiple accounts. Is it possible to have a single keystore file, with multiple keys in it?

Comment (sbaranov) [reply keystore.js:15]: On second thought, I don't have a strong preference for single vs multiple files here.

File: cli.js (hunk 2 above)
Comment (kai0x) [reply cli.js:64]: @sbaranov thanks 👍 do you think we should still have these command inputs here as `key import <mnemomic> <password>` 

or would it be better to just be `key import` and then use `readline-sync` to prompt for the mnemomic and password?

I think the Heroku cli uses the latter more often: https://devcenter.heroku.com/articles/heroku-cli. Heroku uses oclif so I think this is what they use: https://oclif.io/docs/prompting

Comment (kai0x) [reply cli.js:64]: I think the Heroku cli uses the latter more often: https://devcenter.heroku.com/articles/heroku-cli. Heroku uses oclif so I think this is what they use: https://oclif.io/docs/prompting

Comment (sbaranov) [reply cli.js:64]: I think prompt would be the main use case. It seems fine to remove these parameters.

Comment (kai0x) [reply cli.js]: sounds good 👍 I swaped it out to use the `inquirer` npm package. it seems its the same heroku uses, 
Screen Shot 2022-10-12 at 3 29 36 PM

**Comment by kai0x** [comment]
@sbaranov can I get a review on this one? just wanted to merge this one in first to have less rebase conflicts. thanks! :)

File: cli.js [hunk 5]
```diff
@@ -58,6 +60,36 @@ yargs(hideBin(process.argv))
...
     )
+    .command(
+        'key import',
+        'Import a wallet into armada-cli. Private key is not stored. An encrypted keystore is generated from private key and used instead for signing armada transactions',
+        (yargs) => {
+            return yargs
+        },
+        async () => {
+            let responses = await inquirer.prompt([{
+                name: 'address',
```
Comment (sbaranov) [inline cli.js:71]: Should extract the address from the private key, no parameter needed.

Comment (sbaranov) [reply cli.js:71]: Should be something like this: `new ethers.Wallet(privateKey).address`

**Review by sbaranov — APPROVED** [review]

File: cli.js (hunk 5 above)
Comment (kai0x) [reply cli.js]: sounds good! just added this in

**Pull request merged**